addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4454,10 +4463,154 @@ registry.ImageOptimize = ImageHandlerOption.extend({
     // Private
     //--------------------------------------------------------------------------
+    /**
+     * @override
+     */
+    async _applyOptions(update) {
+        const img = await this._super(...arguments);
+        if (img && img.dataset.shape) {
+            await this._applyShape(img.dataset.shape, 'reset');
+        }
+    },
+
+    /**
+     * Applies the shape to the current image by converting the image to base 64 and including it into the svg
+     * @param {string} shape Part of the path to the svg file
+     * @param {*} previewMode state of the current editor
+     * @returns Returns a promise
+     */
+    async _applyShape(shape, previewMode) {
+        const img = this._getImg();
+        if (!shape) {
+            if (!img.dataset.shape) {
+                return;
+            }
+            const svgUrl = img.getAttribute('src');
+            const svgText = await (await fetch(svgUrl)).text();
+            const SVGParser = new DOMParser();
+            const svg = SVGParser.parseFromString(svgText, 'image/svg+xml').documentElement;
+            const svgImage = svg.querySelector('image');
+            img.src = svgImage.getAttribute('xlink:href');
+            img.mimetype = 'image/jpeg';
+            delete img.dataset.shape;
+            return;
+        }
+        const [module, fileName] = shape.split('/');
+        const clipShapeURL = `/${module}/static/image_shapes/${fileName}.svg`;
+        const clipShape = await (await fetch(clipShapeURL)).text();
+        const SVGParser = new DOMParser();
+        const svg = SVGParser.parseFromString(clipShape, 'image/svg+xml').documentElement;
+
+        const base64 = await applyModifications(img);
+
+        const svgImage = svg.querySelector('image');
+        svgImage.setAttribute('xlink:href', base64);
+        svg.setAttribute('width', img.naturalWidth);
+        svg.setAttribute('height', img.naturalHeight);
+        img.dataset.shape = shape;
+        await this._writeSvg(svg.outerHTML, fileName);
+        return this._applyColor(img.dataset.shapeColor, true, previewMode);
+    },
+
+    /**
+     * Applies the correct color to the shape by looking for the default palette's hexcode inside the OG svg files
+     * @param {string} color HEX Code for the color of the background shape
+     * @param {bool} newShape wether it's a new shape or not so we know to look for default colors or previous ones
+     * @param {string} previewMode state of the editor to know which colors to look for
+     */
+    async _applyColor(color, newShape, previewMode) {
+        const img = this._getImg();
+        if (!img.dataset.shape) {
+            return;
+        }
+
+        const fileName = img.dataset.shape.split('/')[1];
+        const clipShapeURL = img.src;
+        let clipShape = await (await fetch(clipShapeURL)).text();
+        let newColor = color && normalizeColor(color);
+        if (newShape) {
+            const DEFAULT_PALETTE = weUtils.DEFAULT_PALETTE;
```

Comment:
```js
const {DEFAULT_PALETTE} = weUtils;
```
(and you might want that just under the import of weUtils at the top of the file)